File: Sorting C/bubbleSort.c
#include<stdio.h>

void bubbleSort(int arr[], int n) {
    int i, j, temp;
    for(i=0; i<n; i++) {
        for(j=i+1; j<n; j++) {
            if(arr[j] < arr[i]) {
                temp = arr[j];
                arr[j] = arr[i];
                arr[i] = temp;
            }
        }

    }

}

int main() {
    int arr[] = {5, 2, 8, 3, 4}, n=5;
    bubbleSort(arr, n);

    for(int i=0; i<n; i++)
        printf("%d ", arr[i]);
    return 0;
}

File: ror/bubbleSort.rb
def bubbleSort(arr)
  size = arr.size
  for i in 0...(size - 1)
    for j in 0...(size - i - 1)
      if arr[j] > arr[j + 1]
        temp = arr[j]
        arr[j] = arr[j + 1]
        arr[j + 1] = temp
      end
    end
  end
  arr
end

arr = [5, 3, 1]
sorted_arr = bubbleSort(arr)
puts sorted_arr.inspect
